### ISSUE TITLE
Pen action (click/draw) delayed, while cursor without lag

### ISSUE THREAD

**Issue by morkr** [linux-surface/linux-surface#1438]
Hello,

I am experiencing a problem with a delay between putting the Pen tip on the screen and actually starting to draw/click. 
In Xournal++ for example, I put the tip on the glass, start a line and the line is visible only after 2 cm or so, while the cursor shows no lag. The line starts earlier when I apply more pressure and when I am afraid that I will puncture the screen next, it starts where it should. I can also put the pen on the glass with low-medium pressure and wait for maybe a sec and then I can draw.
Once the line ist there, I can do everything I want. Pressure sensitivity is working, tilt as well. But when I lift for the next stroke I have the same issue again.
The pressure sensitivity is reacting to the controls for hard/soft in the system settings and I also tested adjusting the pressure curve manually to extremes via dconf-editor.
Same behaviour. First the delay and then everything as expected.
It happens systemwide, as far as I can tell (tested Xournal++, Xournal, file manager, browser, other apps).

I got the device, tested everything working, including the pen, in Windows 11, updated Windows and then deleted Windows and overwrote with Debian 12 with full disk encryption. Testing Debian as live system I was experiencing the issue already, but was hoping the linux-surface kernel would fix it.
Now I wish I had left Windows there and also tested the pen again in Windows after the updates… Not my brightest moment there.

The bug  happens across different kernels and linux distributions (stated below), as well as with both Wayland and X11, so I don't think it is a specific linux-surface kernel issue, but you here seem to have a very good understanding of the internals of these matters, so I thought I could at least ask for a pointer how to debug this. I am afraid it might be a hardware issue. I already tested two different pens (both bought used) which show the same behaviour.
I bought the device (2nd hand from private) for drawing on screen on a linux device, so I am kinda desparate to get it working. Any help is welcome.

Cheers,
morkr

### Environment

- Hardware model: Surface Go 2, 8 GB RAM, 128 GB SSD, Intel® Core™ m3-8100Y × 4, Surface Pen without clip (1776)
- Kernel version: `6.8.6-surface-1 #1 SMP PREEMPT_DYNAMIC Tue Apr 16 12:48:24 UTC 2024 x86_64 GNU/Linux`
The issue is also present on originally installed: `6.1.0-20-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.85-1 (2024-04-11) x86_64 GNU/Linux`
- Distribution: Debian 12 GNOME / standard edition (issue exists in Wayland and X11 both), also tested on live system Ubuntu 24.04 and Pop OS 22.04 

<details><summary>`dmesg` output</summary>

https://notebin.de/?30901a0ce6576c7f#97VGEHV2hitSeU8P4AK33C6m668w6QCwDRkznaWvBR6Q

link due to word limit

</details>

I am also experiencing an issue with sound after getting the cameras to work, which exists only on linux-surface kernel. That will be a separate issue, when I find the time.

**Comment by morkr** [comment]
update: in the meantime I tried booting from a Windows 11 USB and discovered that the pen is definitely working as expected in the Windows menus. So it is not a hardware issue, I guess.
I also tested on Fedora 40 Live system and had the same errant behaviour as with Debian, Ubuntu etc.

Anyone having an idea where I can dig? Any settings for delayed registering of pen action or sth like that?
